quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -161,6 +186,74 @@ impl MappingLeaf {
     }
 }
 
+fn convert_json_val(json_type: JsonType, json_val: JsonValue) -> Option<JsonValue> {
```

Comment:
it's not really a conversion

the match can be simplified with

```
JsonValue::Number(_) if json_type == JsonType::Number => Some(json_val),
```

Reply:
Nice catch. Yes it is not a conversion.